File: client/coresource/src/ru/spbstu/gyboml/core/net/SessionInfo.java
package ru.spbstu.gyboml.core.net;

import lombok.AllArgsConstructor;
import lombok.NoArgsConstructor;
import lombok.NonNull;
import ru.spbstu.gyboml.core.Player;

/**
 * Class repersents one session lobby menu
 * Only for serialization and transfer between server and client.
 * */
@AllArgsConstructor
@NoArgsConstructor
public class SessionInfo {

    // id of related server session
    public @NonNull Integer sessionId;

    // number of free spaces in lobby
    public @NonNull Integer spaces;

    // lobby name
    public @NonNull String name;

    // players
    public Player firstPlayer;
    public Player secondPlayer;

    // is it started
    public @NonNull boolean isStarted;
}

File: client/coresource/src/ru/spbstu/gyboml/core/net/SessionInfo.kt
package ru.spbstu.gyboml.core.net

data class SessionInfo(val id: Int                       = 0,
                       val spaces: Int                   = 0,
                       val name: String                  = "default_session",
                       val firstPlayer: SessionPlayer?   = null,
                       val secondPlayer: SessionPlayer?  = null,
                       val isStarted: Boolean            = false)